core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -500,11 +504,13 @@ class ReplicaManager(val config: KafkaConfig,
               // Delete log and corresponding folders in case replica manager doesn't hold them anymore.
               // This could happen when topic is being deleted while broker is down and recovers.
               stoppedPartitions += topicPartition -> deletePartition
+              if (remoteLogManager.isDefined)
+                partitionsMaybeToDeleteRemote += topicPartition
```

Comment:
When the state of the partition is HostedPartition.None, it indicates that the topic was deleted during broker down and recoveries. Therefore, it can be directly added to 'partitionsMaybeToDeleteRemote'.

Reply:
Hmm, `allPartitions` only stores the partitions with a replica in this broker. So, `HostedPartition.None` only means that the partition doesn't reside in this broker, but the partition could still exist in other brokers in the cluster.

Reply:
Thank you for the explanation, here will additionally verify the `requestLeaderEpoch` information.